azurerm/internal/services/containers/kubernetes_addons.go

```diff
@@ -154,6 +157,20 @@ func schemaKubernetesAddOnProfiles() *schema.Schema {
 						},
 					},
 				},
+
+				"open_service_mesh": {
+					Type:     schema.TypeList,
+					MaxItems: 1,
+					Optional: true,
+					Elem: &schema.Resource{
+						Schema: map[string]*schema.Schema{
+							"enabled": {
+								Type:     schema.TypeBool,
+								Required: true,
+							},
```

Comment:
can we ensure these are set back into the state too? this is done in the flatten function below: https://github.com/terraform-providers/terraform-provider-azurerm/pull/11189/files#diff-c28ad69551916efc44930e639bf392636923aa37f4a68e17cb874edf2c34b101R312-R341